src/Workspaces/Core/Portable/CodeStyle/CodeStyleOption.cs

```diff
@@ -29,10 +35,34 @@ public CodeStyleOption(T value, NotificationOption notification)
         public XElement ToXElement() =>
             new XElement(nameof(CodeStyleOption<T>), // `nameof()` returns just "CodeStyleOption"
                 new XAttribute(nameof(SerializationVersion), SerializationVersion),
-                new XAttribute("Type", typeof(T).Name),
-                new XAttribute(nameof(Value), Value),
+                new XAttribute("Type", GetTypeNameForSerialization()),
+                new XAttribute(nameof(Value), GetValueForSerialization()),
                 new XAttribute(nameof(DiagnosticSeverity), Notification.Value));
 
+        private object GetValueForSerialization()
+        {
+            if (typeof(T) == typeof(bool))
```

Comment:
No.  These are not variables that are being checked.  and "typeof(T)" would only pattern match against System.Type.